integrations/jina/src/haystack_integrations/components/embedders/jina/document_embedder.py

```diff
@@ -57,16 +56,15 @@ def __init__(
         :param meta_fields_to_embed: List of meta fields that should be embedded along with the Document text.
         :param embedding_separator: Separator used to concatenate the meta fields to the Document text.
         """
-
-        api_key = api_key or os.environ.get("JINA_API_KEY")
-        # we check whether api_key is None or an empty string
-        if not api_key:
+        resolved_api_key = api_key.resolve_value()
+        if resolved_api_key is None:
             msg = (
                 "JinaDocumentEmbedder expects an API key. "
                 "Set the JINA_API_KEY environment variable (recommended) or pass it explicitly."
             )
             raise ValueError(msg)
```

Comment:
`resolve_value()` will raise an error if `strict=True`, which it is by default, and the env var is not set, so this is dead code. Let's remove it and directly call the method when updating the headers. Same for the other component too.